User Input questions - New Design

nfmohit opened the pull request:
## Summary

<!-- Please reference the issue this PR addresses in the following list. -->
Addresses issue:

- #5890 

## Relevant technical choices

This PR implements the new design for User Input questions.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

techanvil commented on assets/sass/components/user-input/googlesitekit-user-input-module.scss:
```diff
@@ -64,4 +65,56 @@
+		@media (min-width: $bp-desktop) {
+			align-items: center;
+			border-top: 1px solid $c-mercury;
+			flex-direction: row;
```
Another minor point but I'd suggest `row-reverse` here would make things a bit clearer than specifying `order: 2` below.
```suggestion
			flex-direction: row-reverse;
```

nfmohit replied:
Updated, thank you!

techanvil reviewed (changes requested):
Hi @nfmohit, nice work here. I have a left a few comments, please take a look.

nfmohit commented on assets/js/components/user-input/UserInputQuestionWrapper.js:
```diff
@@ -87,52 +89,55 @@ export default function UserInputQuestionWrapper( props ) {
+			</div>
+			{ isActive && (
+				<div className="googlesitekit-user-input__footer googlesitekit-user-input__buttons">
+					<div className="googlesitekit-user-input__footer-left">
```
Updated. Thank you!

nfmohit commented on assets/js/components/user-input/CancelUserInputButton.js:
```diff
@@ -0,0 +1,48 @@
+/**
+ * Cancel User Input Action.
```
Updated, thank you!

nfmohit commented on assets/sass/components/user-input/googlesitekit-user-input-controls.scss:
```diff
@@ -82,6 +60,19 @@
+		.mdc-text-field--error .mdc-line-ripple--active {
+			background-color: $c-black;
+			opacity: 1;
+			transform: scaleX(1);
```
Must've been a copy-paste mistake. Removed, thank you for noticing!

nfmohit commented on assets/js/components/user-input/UserInputSelectOptions.js:
```diff
@@ -224,28 +252,45 @@ export default function UserInputSelectOptions( {
-								! values.includes( other.trim() ) || ! isActive
-									? '-1'
-									: undefined
+					<div className="googlesitekit-user-input__select-option-text-field">
```
Nice catch, thank you! I've updated this text field to have a fluid-width so that it adapts to the container width, but with a maximum of `250px` (current fixed width).

techanvil replied:
Thanks, that's much better. There is still one slight visual hiccup, which is the placeholder text is truncated at narrow viewports. Here it is at 320px (in the plugin as opposed to Storybook which actually provides less than 320px due to the margins added to the story):

![image](https://user-images.githubusercontent.com/18395600/203538520-17c16ce4-11d5-4d74-99ae-d3f16a91d317.png)

I'd suggest adding the following CSS to provide ellipses in this scenario:

```css
input:placeholder-shown {
	text-overflow: ellipsis;
}
```

Resulting in:

![image](https://user-images.githubusercontent.com/18395600/203539323-220fb1d4-f6b6-4cd7-96c9-fd487f4f1b44.png)

If necessary we can always iterate further on a separate issue...

nfmohit commented on assets/js/components/user-input/UserInputQuestionWrapper.js:
```diff
@@ -87,52 +89,55 @@ export default function UserInputQuestionWrapper( props ) {
+							</Button>
+						) }
+					</div>
+					<div className="googlesitekit-user-input__footer-right">
```
Updated, thanks!

nfmohit commented on assets/sass/components/user-input/googlesitekit-user-input-module.scss:
```diff
@@ -64,4 +65,56 @@
+		padding: 17px 17px 33px;
+
+		@media (min-width: $bp-desktop) {
+			order: 2;
```
Updated, thank you!

nfmohit commented on assets/sass/components/user-input/googlesitekit-user-input-module.scss:
```diff
@@ -64,4 +65,56 @@
+		grid-row: span 2;
+	}
+
+	p.googlesitekit-user-input__select-instruction {
```
Thank you for checking!

If I remove the `p`, styles for `.googlesitekit-plugin p:not([class*=mdc-])` take precedence (screenshot below). Do you think we should adopt a better approach?

![image](https://user-images.githubusercontent.com/20284937/203501316-53a08378-896f-40f9-9fc7-83d36c6efc4a.png)

techanvil replied:
I was just checking as the other classes were not qualified with the tag name. Adding it for precedence is all good, thanks for clarifying!

nfmohit commented:
> Hi @nfmohit, nice work here. I have a left a few comments, please take a look.

Thank you very much for the kind review, @techanvil. I have addressed all the comments and left a follow-up question [here](https://github.com/google/site-kit-wp/pull/6130#discussion_r1030148959). Looking forward to moving this forward.

techanvil reviewed (changes requested):
Hey @nfmohit, thanks for making those changes as well as updating the copy.

Just one small visual issue to address and this looks good to go!

nfmohit commented:
I've added the ellipsis, thank you @techanvil!
